CMakeLists.txt

```diff
@@ -1,253 +1,322 @@
+    list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
+
+    set(HDF5_PREFER_PARALLEL true)
+
+#   Try to find the required dependency
+    find_package(MPI QUIET)
+    find_package(HDF5 QUIET COMPONENTS C HL)
+    find_package(CUDAToolkit QUIET)
+    find_package(OpenMP REQUIRED)
+
+    find_package(Python3 3.8 REQUIRED)
+    if(NOT Python3_FOUND)
+        message(
+            WARNING
+                "We cannot find Python3 and the OPS code-generator needs Python3! Please use -DPython3_EXECUTABLE to specify the path."
+        )
+    endif()
+
+    if(OPS_HIP)
+#       Search for rocm in common locations
+        list(APPEND CMAKE_PREFIX_PATH /opt/rocm/hip /opt/rocm)
+
+#       Find hip
+        find_package(hip)
+
+#       enable_language(HIP) message($ENV{HIP_PLATFORM})
+
+        set(ENV{HIP_PLATFORM} "nvidia")
+#       message($ENV{HIP_PLATFORM})
+
+        set(HIP_FOUND TRUE)
+
+        if({$hip::host} STREQUAL "" OR {$hip::device} STREQUAL "")
+            message(
+                WARNING
+                    "We cannot find the HIP environment. The HIP codes won't work! You might need to use CMAKE_PREFIX_PATH to specify the path for HIP!"
+            )
+            set(HIP_FOUND FALSE)
+        endif()
+    endif()
+
+#   Configure the "include" dir for compiling
+    if(NOT HDF5_FOUND)
+        message(
+            WARNING
+                "We cannot find the parallel HDF5 library. The HDF5 IO routines won't work! Please use -DHDF5_ROOT to specify the path!"
+        )
+    endif()
+
+    if(NOT MPI_FOUND)
+        message(
+            WARNING "We cannot find a MPI environment! The MPI codes won't compile!")
+    endif()
+
+    if(LIBTRID_PATH)
+        include_directories(${LIBTRID_PATH}/include)
+    endif()
+
+    if(CUDAToolkit_FOUND)
+        set(CMAKE_CUDA_COMPILER ${CUDAToolkit_NVCC_EXECUTABLE})
+        if(GPU_ARCH)
+            set(CMAKE_CUDA_ARCHITECTURES
+                ${GPU_ARCH}
+                CACHE STRING "CUDA architectures")
+        else()
+            set(CMAKE_CUDA_ARCHITECTURES
+                70
+                CACHE STRING "CUDA architectures")
+                message(WARNING "Using Volta architecure for compiling CUDA. \nIf different please provide correct GPU architecture using -DGPU_ARCH=XXX  where XXX is compute capability of an architecture!")
+        endif()
+
+        enable_language(CUDA)
+
+    else()
+        message(WARNING "We cannot find a CUDA toolkit! The CUDA codes won't compile!")
+    endif()
+
+    if(USE_NCCL)
+        find_package(NCCL)
+        if(NOT NCCL_FOUND)
+            message(
+                WARNING
+                    "We cannot find NCCL! Tridiagonal library wrapper will be compiled with NCCL"
+            )
+        endif()
+    endif()
+
+#   option(BUILD_OPS_FORTRAN "Build FORTRAN version of OPS" OFF)
+    option(BUILD_OPS_APPS "Build Sample OPS applications" OFF)
+
+#   option(OPS_ERROR_ON_WARNING "Error upon encountering a warning" ON)
```

Comment:
why leave these commented out ones?

Reply:
It is commented only, not sure if it was tested before